clang/lib/Basic/Targets/X86.cpp

```diff
@@ -1763,10 +1823,14 @@ void X86TargetInfo::fillValidTuneCPUList(SmallVectorImpl<StringRef> &Values) con
 }
 
 ArrayRef<const char *> X86TargetInfo::getGCCRegNames() const {
+  if (HasEGPR)
+    return llvm::ArrayRef(ExtendedGCCRegNames);
   return llvm::ArrayRef(GCCRegNames);
```

Comment:
Shouldn't we extend GCCRegNames directly regardless of the target feature?

Reply:
I noticed gcc did so. it reports error without -mapxf when compiling `r16`. reasonable to me.

Reply:
It's a gap. https://www.godbolt.org/z/xx8nrMdYW
But please keep the behavior consistent inside clang

Reply:
Got it. Addressed in [22a1795](https://github.com/llvm/llvm-project/pull/91323/commits/22a17952999aa2b10b8a71c71f7607e37290b973)